include/kernel.h

```diff
@@ -607,13 +607,19 @@ extern FUNC_NORETURN void k_thread_user_mode_enter(k_thread_entry_t entry,
  * previous pool.
  *
  * @param thread Target thread to assign a memory pool for resource requests.
- * @param pool Memory pool to use for resources,
+ * @param heap Heap object to use for resources,
  *             or NULL if the thread should no longer have a memory pool.
  */
-static inline void k_thread_resource_pool_assign(struct k_thread *thread,
+static inline void k_thread_heap_assign(struct k_thread *thread,
+					struct k_heap *heap)
+{
+	thread->resource_pool = heap;
+}
+
+static inline void z_thread_resource_pool_assign(struct k_thread *thread,
```

Comment:
this will break existing users.
better I think to leave this function name as `k_thread_resource_pool_assign` and mark as `__deprecated`